zetaclient/chains/base/observer.go

```diff
@@ -441,13 +460,14 @@ func (ob *Observer) PostVoteInbound(
 		Str(logs.FieldBallotIndex, ballot).
 		Logger()
 
-	switch {
-	case err != nil:
+	if err != nil {
```

Comment:
This is a mistake from previous code but the error check should come before the `logger = logger.With()` above